resource_vultr_server.go

```diff
@@ -129,6 +139,8 @@ func resourceVultrServerCreate(d *schema.ResourceData, meta interface{}) error {
 		IPXEChainURL: d.Get("ipxe_chain_url").(string),
 		ISO:          d.Get("iso_id").(int),
 		UserData:     d.Get("user_data").(string),
+		Hostname:     d.Get("hostname").(string),
```

Comment:
Since `hostname` and `tag` are optional you have to check for their existence first. Like what is done for `ipv6`.